### DISCUSSION TITLE
Some confusion about ADVolumeJunction1PhaseUserObject in THM

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#26889] [Q&A Modules: Thermal Hydraulics]
Hi @GiudGiud and @jasondhales,

I noticed that we just calculate the `S_loss` at` c==0` in `ADVolumeJunction1PhaseUserObject`, which is the first inlet section in this junction. Why do not we calculate the `S_loss` of other inlet sections (if this junction contains more than one inlet section)? I'm really confused about it. Could you please tell me?

Thanks!

**Comment by GiudGiud** [comment]
@joshuahansel

**Comment by joshuahansel** [comment]
The term is not meant to be a per-connection term. It looks like we've just chosen the first connection arbitrarily for the reference values. I agree that something needs to be done - we shouldn't have the property the that the solution depends on the ordering of the connections. @licharlot What do you think is the right solution? I guess we want some consistency with legacy systems codes? Do any of those have per-connection form loss terms, or should we do something like use an average inlet condition?

**Comment by licharlot** [comment]
Correct, the loss coefficient in the volume junction is associated with the flow conditions of the first connection. The associated area can be user defined, or is the one of the first connection only.  Most codes have a loss coefficient associated with each connection, forward and reverse loss coefficients.

**Reply by SomnusYu** [reply]
That's to say, we truly need to calculate the loss coefficient of each connection, but THM just supports the loss coefficient of the first connection (just for example). We still need to improve THM to include all connections. Is that right?

**Reply by licharlot** [reply]
Yes, we'd have to go back to the volume junction formulation and see how we can integrate this.

**Reply by SomnusYu** [reply]
Thanks!

**Reply by joshuahansel** [reply]
Yeah, formulation-wise, it seems straightforward. It's mostly just a matter of transitioning the new "interface" with some temporary backwards compatibility.

**Reply by SomnusYu** [reply]
By the way, I would like to know if it is two-phase flow, what the momentum residual should be. I just write these terms according to single phase flow, could you please tell me if it is correct?

For single phase, it is:
```
  _residual[VolumeJunction1Phase::RHOUV_INDEX] -=
      di(0) * din * _flux[c][THM3Eqn::CONS_VAR_RHOUA] - pJ * ni(0) * _A[0];
  _residual[VolumeJunction1Phase::RHOVV_INDEX] -=
      di(1) * din * _flux[c][THM3Eqn::CONS_VAR_RHOUA] - pJ * ni(1) * _A[0];
  _residual[VolumeJunction1Phase::RHOWV_INDEX] -=
      di(2) * din * _flux[c][THM3Eqn::CONS_VAR_RHOUA] - pJ * ni(2) * _A[0];
```

**For two phase:**
```
  _residual[VolumeJunction2Phase::VEL_LIQUID_U_INDEX] -=
                        di(0) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_LIQ] - (1.0 - alpha_J)* pJ * ni(0) * _A[0];
  _residual[VolumeJunction2Phase::VEL_LIQUID_V_INDEX] -=
                        di(1) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_LIQ] -  (1.0 - alpha_J)* pJ * ni(1) * _A[0];
  _residual[VolumeJunction2Phase::VEL_LIQUID_W_INDEX] -=
                        di(2) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_LIQ] -  (1.0 - alpha_J)* pJ * ni(2) * _A[0];
  _residual[VolumeJunction2Phase::VEL_VAPOUR_U_INDEX] -=
                        di(0) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_VAP] - alpha_J* pJ * ni(0) * _A[0];
  _residual[VolumeJunction2Phase::VEL_VAPOUR_V_INDEX] -=
                        di(1) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_VAP] - alpha_J* pJ * ni(1) * _A[0];
  _residual[VolumeJunction2Phase::VEL_VAPOUR_W_INDEX] -=
                        di(2) * din * _flux[c][THM6Eqn::CONS_VAR_VEL_VAP] - alpha_J* pJ * ni(2) * _A[0];
```

**Reply by joshuahansel** [reply]
Looks right, assuming your `alpha` is the void fraction, not `1 - alpha`.

**Reply by SomnusYu** [reply]
Yes, alpha is the void fraction. Thanks a lot!
